Objects/complexobject.c

```diff
@@ -970,9 +1030,8 @@ complex_new_impl(PyTypeObject *type, PyObject *r, PyObject *i)
         (nbr->nb_float == NULL && nbr->nb_index == NULL && !PyComplex_Check(r)))
     {
         PyErr_Format(PyExc_TypeError,
-                     "complex() first argument must be a string or a number, "
-                     "not '%.200s'",
-                     Py_TYPE(r)->tp_name);
+                     "complex() argument 'real' must be a real number, not %T",
+                     r);
         if (own_r) {
```

Comment:
Actually, ``!own_r`` condition here is inaccessible.  If we are here - try_complex_special_method() call above was unsuccessful.

See #109642.  Maybe it worth to include constructor-related changes from that pr.

Reply:
I'll see.